--- libpano12-2.7.0.14/filter.h	2005-11-16 12:28:55.000000000 +0000
+++ filter.h	2005-12-17 10:50:27.000000000 +0000
@@ -901,9 +901,15 @@ extern sPrefs			*gsPrPtr;
 	#define		read_bin			"rb"
 	#define		read_write_text			"rw"
 	#define		append_bin			"ab"
-	#define		p2cstr( x )	
-	#define		c2pstr( x )
-															
+    
+    // Ippei hack. OSX with GCC+ANSI mode.
+    #ifdef MAC_OS_X_VERSION_10_4
+        // MacOSX 10.4 has those functions predefined in Carbon API.
+        #include <Carbon/Carbon.h> // CoreServices/TextUtils.h
+    #else
+        #define		p2cstr( x )	
+        #define		c2pstr( x )
+    #endif
 
 #endif
 
