--- ../wxMac-2.6.1/src/mac/carbon/dirdlg.cpp	2005-05-30 12:29:00.000000000 +0100
+++ src/mac/carbon/dirdlg.cpp	2005-06-06 21:43:24.000000000 +0100
@@ -50,18 +50,16 @@
 
 int wxDirDialog::ShowModal()
 {
-    NavDialogOptions        mNavOptions;
-    NavObjectFilterUPP        mNavFilterUPP = NULL;
-    NavPreviewUPP            mNavPreviewUPP = NULL ;
-    NavReplyRecord            mNavReply;
-    AEDesc*                    mDefaultLocation = NULL ;
-    bool                    mSelectDefault = false ;
+    NavDialogCreationOptions   mNavOptions;
+//    NavObjectFilterUPP         mNavFilterUPP = NULL;
+//    NavPreviewUPP              mNavPreviewUPP = NULL ;
+    NavReplyRecord             mNavReply;
+//    AEDesc*                    mDefaultLocation = NULL ;
+//    bool                       mSelectDefault = false ;
     
-    ::NavGetDefaultDialogOptions(&mNavOptions);
+    ::NavGetDefaultDialogCreationOptions(&mNavOptions);
 
-    mNavFilterUPP    = nil;
-    mNavPreviewUPP    = nil;
-    mSelectDefault    = false;
+//    mSelectDefault    = false;
     mNavReply.validRecord                = false;
     mNavReply.replacing                    = false;
     mNavReply.isStationery                = false;
@@ -75,22 +73,33 @@
     //   that's displayed when the dialog
     //   first appears
     
-    if ( mDefaultLocation ) {
+//    if ( mDefaultLocation ) {
         
-        if (mSelectDefault) {
-            mNavOptions.dialogOptionFlags |= kNavSelectDefaultLocation;
-        } else {
-            mNavOptions.dialogOptionFlags &= ~kNavSelectDefaultLocation;
-        }
-    }
-    
-    OSErr err = ::NavChooseFolder(
-                        mDefaultLocation,
-                        &mNavReply,
-                        &mNavOptions,
-                        NULL,
-                        mNavFilterUPP,
-                        0L);                            // User Data
+//        if (mSelectDefault) {
+//            mNavOptions.dialogOptionFlags |= kNavSelectDefaultLocation;
+//        } else {
+            mNavOptions.optionFlags &= ~kNavSelectDefaultLocation;
+//        }
+//    }
+    
+    wxMacCFStringHolder message(m_message, m_font.GetEncoding());
+    mNavOptions.message = message;
+    
+    NavDialogRef dialog;
+    OSErr err = NavCreateChooseFolderDialog(&mNavOptions, NULL, NULL, NULL, &dialog);
+    
+    if (err == noErr)
+        err = ::NavDialogRun(dialog);
+    
+    err = ::NavDialogGetReply(dialog, &mNavReply);
+    
+//    OSErr err = ::NavChooseFolder(
+//                        mDefaultLocation,
+//                        &mNavReply,
+//                        &mNavOptions,
+//                        NULL,
+//                        mNavFilterUPP,
+//                        0L);                            // User Data
     
     if ( (err != noErr) && (err != userCanceledErr) ) {
         m_path = wxT("") ;
