--- ../wxMac-2.6.2/src/mac/corefoundation/utilsexc_base.cpp	2005-11-28 20:50:01.000000000 +0000
+++ src/mac/corefoundation/utilsexc_base.cpp	2005-11-28 20:49:44.000000000 +0000
@@ -47,10 +47,17 @@
         return errorCode ;
     }
     
-    CFURLRef cfurlApp = CFURLCreateWithString(
+//    CFURLRef cfurlApp = CFURLCreateWithString(
+//            kCFAllocatorDefault,
+//            wxMacCFStringHolder(*argv++, wxLocale::GetSystemEncoding()),
+//            NULL);
+        
+CFURLRef cfurlApp = CFURLCreateWithFileSystemPath (
             kCFAllocatorDefault,
             wxMacCFStringHolder(*argv++, wxLocale::GetSystemEncoding()),
-            NULL);
+            kCFURLPOSIXPathStyle,
+            false);
+        
     wxASSERT(cfurlApp);
 
     CFBundleRef cfbApp = CFBundleCreate(kCFAllocatorDefault, cfurlApp);
