diff -w -uNr vigra-1.2.0-orig/include/vigra/numerictraits.hxx vigra-1.2.0/include/vigra/numerictraits.hxx
--- vigra-1.2.0-orig/include/vigra/numerictraits.hxx	2003-08-07 02:16:31.000000000 +0200
+++ vigra-1.2.0/include/vigra/numerictraits.hxx	2004-05-10 08:31:07.000000000 +0200
@@ -313,6 +313,7 @@
     typedef Error_NumericTraits_not_specialized_for_this_case isScalar;
     typedef Error_NumericTraits_not_specialized_for_this_case isIntegral;
     typedef Error_NumericTraits_not_specialized_for_this_case isOrdered;
+    typedef Error_NumericTraits_not_specialized_for_this_case isSigned;
 };
 
 #ifndef NO_BOOL
@@ -325,6 +326,7 @@
     typedef VigraTrueType isIntegral;
     typedef VigraTrueType isScalar;
     typedef VigraTrueType isOrdered;
+    typedef VigraFalseType isSigned;
     
     static bool zero() { return false; }
     static bool one() { return true; }
@@ -359,6 +361,7 @@
     typedef VigraTrueType isIntegral;
     typedef VigraTrueType isScalar;
     typedef VigraTrueType isOrdered;
+    typedef VigraTrueType isSigned;
     
     static signed char zero() { return 0; }
     static signed char one() { return 1; }
@@ -393,6 +396,7 @@
     typedef VigraTrueType isIntegral;
     typedef VigraTrueType isScalar;
     typedef VigraTrueType isOrdered;
+    typedef VigraFalseType isSigned;
     
     static unsigned char zero() { return 0; }
     static unsigned char one() { return 1; }
@@ -426,6 +430,7 @@
     typedef VigraTrueType isIntegral;
     typedef VigraTrueType isScalar;
     typedef VigraTrueType isOrdered;
+    typedef VigraTrueType isSigned;
     
     static short int zero() { return 0; }
     static short int one() { return 1; }
@@ -463,6 +468,7 @@
     typedef VigraTrueType isIntegral;
     typedef VigraTrueType isScalar;
     typedef VigraTrueType isOrdered;
+    typedef VigraFalseType isSigned;
 
     static short unsigned int zero() { return 0; }
     static short unsigned int one() { return 1; }
@@ -497,6 +503,7 @@
     typedef VigraTrueType isIntegral;
     typedef VigraTrueType isScalar;
     typedef VigraTrueType isOrdered;
+    typedef VigraTrueType isSigned;
 
     static int zero() { return 0; }
     static int one() { return 1; }
@@ -530,6 +537,7 @@
     typedef VigraTrueType isIntegral;
     typedef VigraTrueType isScalar;
     typedef VigraTrueType isOrdered;
+    typedef VigraFalseType isSigned;
     
     static unsigned int zero() { return 0; }
     static unsigned int one() { return 1; }
@@ -563,6 +571,7 @@
     typedef VigraTrueType isIntegral;
     typedef VigraTrueType isScalar;
     typedef VigraTrueType isOrdered;
+    typedef VigraTrueType isSigned;
     
     static long zero() { return 0; }
     static long one() { return 1; }
@@ -596,6 +605,7 @@
     typedef VigraTrueType isIntegral;
     typedef VigraTrueType isScalar;
     typedef VigraTrueType isOrdered;
+    typedef VigraFalseType isSigned;
     
     static unsigned long zero() { return 0; }
     static unsigned long one() { return 1; }
@@ -629,6 +639,7 @@
     typedef VigraFalseType isIntegral;
     typedef VigraTrueType isScalar;
     typedef VigraTrueType isOrdered;
+    typedef VigraTrueType isSigned;
     
     static float zero() { return 0.0; }
     static float one() { return 1.0; }
@@ -653,6 +664,7 @@
     typedef VigraFalseType isIntegral;
     typedef VigraTrueType isScalar;
     typedef VigraTrueType isOrdered;
+    typedef VigraTrueType isSigned;
     
     static double zero() { return 0.0; }
     static double one() { return 1.0; }
@@ -677,6 +689,7 @@
     typedef VigraFalseType isIntegral;
     typedef VigraTrueType isScalar;
     typedef VigraTrueType isOrdered;
+    typedef VigraTrueType isSigned;
     
     static long double zero() { return 0.0; }
     static long double one() { return 1.0; }
diff -w -uNr vigra-1.2.0-orig/include/vigra/rgbvalue.hxx vigra-1.2.0/include/vigra/rgbvalue.hxx
--- vigra-1.2.0-orig/include/vigra/rgbvalue.hxx	2003-08-07 02:16:32.000000000 +0200
+++ vigra-1.2.0/include/vigra/rgbvalue.hxx	2004-05-16 16:04:32.000000000 +0200
@@ -319,6 +319,7 @@
     typedef RGBValue<typename NumericTraits<T>::RealPromote> RealPromote;
 
     typedef typename NumericTraits<T>::isIntegral isIntegral;
+    typedef typename NumericTraits<T>::isSigned isSigned;
     typedef VigraFalseType isScalar;
     typedef VigraFalseType isOrdered;
 
@@ -378,6 +379,7 @@
     typedef RGBValue<NumericTraits<T>::Promote> Promote;\
     typedef RGBValue<NumericTraits<T>::RealPromote> RealPromote;\
     typedef NumericTraits<T>::isIntegral isIntegral;\
+    typedef NumericTraits<T>::isSigned isSigned;\
     typedef VigraFalseType isScalar;\
     typedef VigraFalseType isOrdered;\
     \
diff -w -uNr vigra-1.2.0-orig/include/vigra/stdimage.hxx vigra-1.2.0/include/vigra/stdimage.hxx
--- vigra-1.2.0-orig/include/vigra/stdimage.hxx	2003-08-07 02:16:33.000000000 +0200
+++ vigra-1.2.0/include/vigra/stdimage.hxx	2004-06-20 14:14:39.000000000 +0200
@@ -88,8 +88,18 @@
     */
 typedef BasicImage<unsigned char> BImage;
 
-VIGRA_DEFINE_ITERATORTRAITS(short, StandardValueAccessor, StandardConstValueAccessor)
+VIGRA_DEFINE_ITERATORTRAITS(unsigned short, vigra::StandardValueAccessor, vigra::StandardConstValueAccessor)
+
+    /** Unsigned Short (16-bit unsigned) image.
+        It uses \ref vigra::BasicImageIterator and \ref vigra::StandardAccessor and
+        their const counterparts to access the data.
+
+        <b>\#include</b> "<a href="stdimage_8hxx-source.html">vigra/stdimage.hxx</a>"<br>
+        Namespace: vigra
+    */
+typedef BasicImage<unsigned short> USImage;
 
+VIGRA_DEFINE_ITERATORTRAITS(short, StandardValueAccessor, StandardConstValueAccessor)
 
     /** Short integer (16-bit signed) image.
         It uses \ref vigra::BasicImageIterator and \ref vigra::StandardAccessor and 
@@ -100,6 +110,17 @@
     */
 typedef BasicImage<short> SImage;
 
+VIGRA_DEFINE_ITERATORTRAITS(unsigned int, StandardValueAccessor, StandardConstValueAccessor)
+
+    /** Unsigned Integer (32-bit unsigned) image.
+        It uses \ref vigra::BasicImageIterator and \ref vigra::StandardAccessor and
+        their const counterparts to access the data.
+
+        <b>\#include</b> "<a href="stdimage_8hxx-source.html">vigra/stdimage.hxx</a>"<br>
+        Namespace: vigra
+    */
+typedef BasicImage<unsigned int> UIImage;
+
 VIGRA_DEFINE_ITERATORTRAITS(int, StandardValueAccessor, StandardConstValueAccessor)
 
     /** Integer (32-bit signed) image.
@@ -145,6 +166,42 @@
     */
 typedef BasicImage<RGBValue<unsigned char> > BRGBImage;
 
+VIGRA_DEFINE_ITERATORTRAITS(RGBValue<unsigned short>, RGBAccessor, RGBAccessor)
+
+    /** Unsigned short (3x 16-bit unsigned) RGB image.
+        The pixel type is \ref vigra::RGBValue "vigra::RGBValue<unsigned short>".
+        It uses \ref vigra::BasicImageIterator and \ref vigra::RGBAccessor and
+        their const counterparts to access the data.
+
+        <b>\#include</b> "<a href="stdimage_8hxx-source.html">vigra/stdimage.hxx</a>"<br>
+        Namespace: vigra
+    */
+typedef BasicImage<RGBValue<unsigned short> > USRGBImage;
+
+VIGRA_DEFINE_ITERATORTRAITS(RGBValue<short>, RGBAccessor, RGBAccessor)
+
+    /** Unsigned short (3x 16-bit signed) RGB image.
+        The pixel type is \ref vigra::RGBValue "vigra::RGBValue<short>".
+        It uses \ref vigra::BasicImageIterator and \ref vigra::RGBAccessor and
+        their const counterparts to access the data.
+
+        <b>\#include</b> "<a href="stdimage_8hxx-source.html">vigra/stdimage.hxx</a>"<br>
+        Namespace: vigra
+    */
+typedef BasicImage<RGBValue<short> > SRGBImage;
+
+VIGRA_DEFINE_ITERATORTRAITS(RGBValue<unsigned int>, RGBAccessor, RGBAccessor)
+
+    /** Unsigned Integer (3x 32-bit unsigned) RGB image.
+        The pixel type is \ref vigra::RGBValue "RGBValue<unsigned int>".
+        It uses \ref vigra::BasicImageIterator and \ref vigra::RGBAccessor and
+        their const counterparts to access the data.
+
+        <b>\#include</b> "<a href="stdimage_8hxx-source.html">vigra/stdimage.hxx</a>"<br>
+        Namespace: vigra
+    */
+typedef BasicImage<RGBValue<unsigned int> > UIRGBImage;
+
 VIGRA_DEFINE_ITERATORTRAITS(RGBValue<int>, RGBAccessor, RGBAccessor)
 
     /** Integer (3x 32-bit signed) RGB image.
@@ -181,6 +238,53 @@
     */
 typedef BasicImage<RGBValue<double> > DRGBImage;
 
+#define VIGRA_PIXELTYPE TinyVector<unsigned char, 2>
+VIGRA_DEFINE_ITERATORTRAITS(VIGRA_PIXELTYPE, VectorAccessor, VectorAccessor)
+#undef VIGRA_PIXELTYPE
+#define VIGRA_PIXELTYPE TinyVector<unsigned char, 3>
+VIGRA_DEFINE_ITERATORTRAITS(VIGRA_PIXELTYPE, VectorAccessor, VectorAccessor)
+#undef VIGRA_PIXELTYPE
+#define VIGRA_PIXELTYPE TinyVector<unsigned char, 4>
+VIGRA_DEFINE_ITERATORTRAITS(VIGRA_PIXELTYPE, VectorAccessor, VectorAccessor)
+#undef VIGRA_PIXELTYPE
+
+#define VIGRA_PIXELTYPE TinyVector<unsigned short, 2>
+VIGRA_DEFINE_ITERATORTRAITS(VIGRA_PIXELTYPE, VectorAccessor, VectorAccessor)
+#undef VIGRA_PIXELTYPE
+#define VIGRA_PIXELTYPE TinyVector<unsigned short, 3>
+VIGRA_DEFINE_ITERATORTRAITS(VIGRA_PIXELTYPE, VectorAccessor, VectorAccessor)
+#undef VIGRA_PIXELTYPE
+#define VIGRA_PIXELTYPE TinyVector<unsigned short, 4>
+VIGRA_DEFINE_ITERATORTRAITS(VIGRA_PIXELTYPE, VectorAccessor, VectorAccessor)
+#undef VIGRA_PIXELTYPE
+#define VIGRA_PIXELTYPE TinyVector<short, 2>
+VIGRA_DEFINE_ITERATORTRAITS(VIGRA_PIXELTYPE, VectorAccessor, VectorAccessor)
+#undef VIGRA_PIXELTYPE
+#define VIGRA_PIXELTYPE TinyVector<short, 3>
+VIGRA_DEFINE_ITERATORTRAITS(VIGRA_PIXELTYPE, VectorAccessor, VectorAccessor)
+#undef VIGRA_PIXELTYPE
+#define VIGRA_PIXELTYPE TinyVector<short, 4>
+VIGRA_DEFINE_ITERATORTRAITS(VIGRA_PIXELTYPE, VectorAccessor, VectorAccessor)
+#undef VIGRA_PIXELTYPE
+#define VIGRA_PIXELTYPE TinyVector<int, 2>
+VIGRA_DEFINE_ITERATORTRAITS(VIGRA_PIXELTYPE, VectorAccessor, VectorAccessor)
+#undef VIGRA_PIXELTYPE
+#define VIGRA_PIXELTYPE TinyVector<int, 3>
+VIGRA_DEFINE_ITERATORTRAITS(VIGRA_PIXELTYPE, VectorAccessor, VectorAccessor)
+#undef VIGRA_PIXELTYPE
+#define VIGRA_PIXELTYPE TinyVector<int, 4>
+VIGRA_DEFINE_ITERATORTRAITS(VIGRA_PIXELTYPE, VectorAccessor, VectorAccessor)
+#undef VIGRA_PIXELTYPE
+#define VIGRA_PIXELTYPE TinyVector<unsigned int, 2>
+VIGRA_DEFINE_ITERATORTRAITS(VIGRA_PIXELTYPE, VectorAccessor, VectorAccessor)
+#undef VIGRA_PIXELTYPE
+#define VIGRA_PIXELTYPE TinyVector<unsigned int, 3>
+VIGRA_DEFINE_ITERATORTRAITS(VIGRA_PIXELTYPE, VectorAccessor, VectorAccessor)
+#undef VIGRA_PIXELTYPE
+#define VIGRA_PIXELTYPE TinyVector<unsigned int, 4>
+VIGRA_DEFINE_ITERATORTRAITS(VIGRA_PIXELTYPE, VectorAccessor, VectorAccessor)
+#undef VIGRA_PIXELTYPE
+
 #define VIGRA_PIXELTYPE TinyVector<float, 2>
 VIGRA_DEFINE_ITERATORTRAITS(VIGRA_PIXELTYPE, VectorAccessor, VectorAccessor) 
 #undef VIGRA_PIXELTYPE 
diff -w -uNr vigra-1.2.0-orig/include/vigra/tinyvector.hxx vigra-1.2.0/include/vigra/tinyvector.hxx
--- vigra-1.2.0-orig/include/vigra/tinyvector.hxx	2003-08-07 02:16:34.000000000 +0200
+++ vigra-1.2.0/include/vigra/tinyvector.hxx	2004-05-10 08:31:09.000000000 +0200
@@ -798,6 +798,7 @@
     typedef TinyVector<typename NumericTraits<T>::RealPromote, SIZE> RealPromote;
 
     typedef typename NumericTraits<T>::isIntegral isIntegral;
+    typedef typename NumericTraits<T>::isSigned isSigned;
     typedef VigraFalseType isScalar;
     typedef VigraFalseType isOrdered;
 
@@ -851,6 +852,7 @@
     typedef TinyVector<typename NumericTraits<T>::RealPromote, SIZE> RealPromote;
 
     typedef typename NumericTraits<T>::isIntegral isIntegral;
+    typedef typename NumericTraits<T>::isSigned isSigned;
     typedef VigraFalseType isScalar;
     typedef VigraFalseType isOrdered;
 };
@@ -914,6 +916,7 @@
     typedef TinyVector<NumericTraits<T>::Promote, SIZE> Promote;\
     typedef TinyVector<NumericTraits<T>::RealPromote, SIZE> RealPromote;\
     typedef NumericTraits<T>::isIntegral isIntegral;\
+    typedef NumericTraits<T>::isSigned isSigned;\
     typedef VigraFalseType isScalar;\
     typedef VigraFalseType isOrdered;\
     \
