diff -Naur wxMac-2.8.10-orig/src/mac/carbon/fontdlgosx.mm wxMac-2.8.10/src/mac/carbon/fontdlgosx.mm
--- wxMac-2.8.10-orig/src/mac/carbon/fontdlgosx.mm	2009-03-06 07:23:14.000000000 -0500
+++ wxMac-2.8.10/src/mac/carbon/fontdlgosx.mm	2009-10-14 22:31:50.000000000 -0400
@@ -169,7 +169,11 @@
 
     NSModalSession session = [NSApp beginModalSessionForWindow:fontPanel];
 
-    [NSApp runModalSession:session];
+    for (;;)
+    {
+        if ([NSApp runModalSession:session] != NSRunContinuesResponse)
+            break;
+    }
 
     [NSApp endModalSession:session];
 
