--- pngconf-bk.h
+++ pngconf.h
@@ -1113,10 +1113,12 @@
  * want to have unsigned int for png_uint_32 instead of unsigned long.
  */
 
-typedef unsigned long png_uint_32;
-typedef long png_int_32;
-typedef unsigned short png_uint_16;
-typedef short png_int_16;
+#include <sys/types.h>
+typedef u_int32_t png_uint_32;
+typedef int32_t png_int_32;
+typedef u_int16_t png_uint_16;
+typedef int16_t png_int_16;
+
 typedef unsigned char png_byte;
 
 /* This is usually size_t.  It is typedef'ed just in case you need it to
