diff --git a/build/CMakeModules/FindGLIB2.cmake b/build/CMakeModules/FindGLIB2.cmake
index 7c39093..b414c72 100644
--- a/build/CMakeModules/FindGLIB2.cmake
+++ b/build/CMakeModules/FindGLIB2.cmake
@@ -1,13 +1,13 @@
-IF (NOT MSVC)
-    INCLUDE(FindPkgConfig)
-    PKG_SEARCH_MODULE( GLIB2 REQUIRED glib-2.0 )
-    IF(WIN32 AND NOT BUILD_STATIC)
-        FIND_FILE(GLIB2_DLL 
-                NAMES glib-2.dll glib-2-vs9.dll libglib-2.0-0.dll
-                PATHS "${GLIB2_LIBRARY_DIRS}/../bin"
-                NO_SYSTEM_ENVIRONMENT_PATH)
-    ENDIF()
-ENDIF()
+# IF (NOT MSVC)
+#     INCLUDE(FindPkgConfig)
+#     PKG_SEARCH_MODULE( GLIB2 REQUIRED glib-2.0 )
+#     IF(WIN32 AND NOT BUILD_STATIC)
+#         FIND_FILE(GLIB2_DLL 
+#                 NAMES glib-2.dll glib-2-vs9.dll libglib-2.0-0.dll
+#                 PATHS "${GLIB2_LIBRARY_DIRS}/../bin"
+#                 NO_SYSTEM_ENVIRONMENT_PATH)
+#     ENDIF()
+# ENDIF()
 
 IF (NOT GLIB2_FOUND OR NOT PKG_CONFIG_FOUND)
     FIND_PATH(GLIB2_GLIB2CONFIG_INCLUDE_PATH
@@ -19,6 +19,7 @@ IF (NOT GLIB2_FOUND OR NOT PKG_CONFIG_FOUND)
           /opt/local/lib
           ${GLIB2_BASE_DIR}/lib
           ${CMAKE_LIBRARY_PATH}
+          ${CMAKE_SOURCE_DIR}/../lib
         PATH_SUFFIXES glib-2.0/include
     )
 
@@ -30,6 +31,7 @@ IF (NOT GLIB2_FOUND OR NOT PKG_CONFIG_FOUND)
             /usr/include
             /opt/local/include
             ${GLIB2_BASE_DIR}/include
+            ${CMAKE_SOURCE_DIR}/../include
         PATH_SUFFIXES gtk-2.0 glib-2.0 glib20 
     )
 
@@ -41,6 +43,7 @@ IF (NOT GLIB2_FOUND OR NOT PKG_CONFIG_FOUND)
             /usr/lib64
             /opt/local/lib
             ${GLIB2_BASE_DIR}/lib
+            ${CMAKE_SOURCE_DIR}/../lib
     )
     
     IF(GLIB2_GLIB2CONFIG_INCLUDE_PATH AND GLIB2_INCLUDE_DIRS AND GLIB2_LIBRARIES)
